portfolio/src/main/java/com/google/sps/data/TimeRangeGroupArrayList.java

```diff
@@ -0,0 +1,121 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/** Models an implementation of the TimeRangeGroup model using ArrayList. */
+public class TimeRangeGroupArrayList implements TimeRangeGroup {
+
+  public List<TimeRange> allTimeRanges;
+  private Comparator<TimeRange> comparator;
+  private boolean ascending;
+
+  /**
+   * Adds all the input time ranges to the list of all time ranges. Also sorts the list of all
+   * ranges in the constructor.
+   */
+  public TimeRangeGroupArrayList(
+      Collection<TimeRange> timeRanges, Comparator<TimeRange> comparator, boolean ascending) {
```

Comment:
Here as well, I don't think we need the `ascending` option.

Reply:
Also I don't think the comparator param is necessary as well. I think the constructor can just take in the collection.

Rationale:
1. Caller can sort the collection as they see fit.
2. If you think this class should perform sorting, I think it should do so using some reasonable sorting approach, which does not need to be specified by the caller.

Reply:
Updated.

Reply:
Hmmm but the class has an `addTimeRange` method so it will have to be able to sort by itself, won't it? Also I don't think it makes sense for this class to try to choose a sort order when the caller has already sorted the collection on the way in. I think it makes sense to keep the `Comparator` as part of this class.

Reply:
I think it would also be fine if you *do* want to just choose a sort order in here and stick with it, but in that case there should be no guarantee that the sort order of the list as it's passed in is preserved at all.